Sources/BraveWallet/Crypto/Stores/NFTDetailStore.swift

```diff
@@ -41,6 +32,17 @@ struct NFTMetadata: Codable, Equatable {
     self.name = name
     self.description = description
   }
+
+  func httpfyIpfsUrl(ipfsApi: IpfsAPI?) -> NFTMetadata {
+    if ipfsApi != nil {
+      if self.imageURLString != nil && self.imageURLString!.hasPrefix("ipfs://") {
+        if let url = URL(string: self.imageURLString!) {
+          return NFTMetadata(imageURLString: ipfsApi?.resolveGatewayUrl(for: url)?.absoluteString, name: self.name, description: self.description)
+        }
+      }
+    }
+    return NFTMetadata(imageURLString: self.imageURLString, name: self.name, description: self.description)
```

Comment:
You can use `if let` or `guard let` to unwrap optionals in Swift to avoid force unwraps with `!` and checking for nil:

```suggestion
    guard let ipfsApi,
          let imageURLString,
          imageURLString.hasPrefix("ipfs://"),
          let url = URL(string: imageURLString)
    else {
      return NFTMetadata(imageURLString: self.imageURLString, name: self.name, description: self.description)
    }
    return NFTMetadata(
      imageURLString: ipfsApi.resolveGatewayUrl(for: url)?.absoluteString,
      name: self.name,
      description: self.description
    )
```